src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,39 @@
 package practice;
 
-public class CandidateValidator {
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
     //write your code here
+    private static final int MIN_AGE = 35;
+    private static final String REQUIRED_NATIONALITY = "Ukrainian";
+    private static final int MIN_PERIOD_LIVING_IN_UKR = 10;
+    private static final String PERIOD_DATA_SEPARATOR = "-";
+    private static final int INDEX_OF_FROM_YEAR = 0;
+    private static final int INDEX_OF_TO_YEAR = 1;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        if (candidate.getAge() < MIN_AGE) {
+            return false;
+        }
+        if (!candidate.isAllowedToVote()) {
+            return false;
+        }
+        if (!candidate.getNationality().equals(REQUIRED_NATIONALITY)) {
+            return false;
+        }
+        if (!isValidPeriodOfLivingInUkr(candidate.getPeriodsInUkr())) {
+            return false;
+        }
+        return true;
+    }
```

Comment:
We can do it more simple
```suggestion
    return candidate.getAge() < MIN_AGE 
                         && !candidate.isAllowedToVote()
                         && !candidate.getNationality().equals(REQUIRED_NATIONALITY)
                         && (!isValidPeriodOfLivingInUkr(candidate.getPeriodsInUkr();
```